modules/claims_api/app/models/claims_api/power_of_attorney.rb

```diff
@@ -16,6 +17,70 @@ class PowerOfAttorney < ApplicationRecord
     before_validation :set_md5
     validates :md5, uniqueness: true
 
+    def sign_pdf
+      signatures = convert_signatures_to_images
+      page_1_path = insert_signatures(1, signatures[:veteran], signatures[:representative])
+      page_2_path = insert_signatures(2, signatures[:veteran], signatures[:representative])
+      { page1: page_1_path, page2: page_2_path }
+    end
+
+    def convert_signatures_to_images
+      {
+        veteran: convert_base64_data_to_image('veteran'),
+        representative: convert_base64_data_to_image('representative')
+      }
+    end
+
+    def convert_base64_data_to_image(signature)
+      path = "tmp/#{signature}_#{id}_signature_b64.png"
+
+      File.open(path, 'wb') do |f|
+        f.write(Base64.decode64(form_data.dig('signatures', signature)))
+      end
+      path
+    end
+
+    def insert_signatures(page, veteran_signature, representative_signature)
+      pdf_path = Rails.root.join('modules', 'claims_api', 'config', 'pdf_templates', "21-22A-#{page}.pdf")
+      stamp_path = "#{::Common::FileHelpers.random_file_path}.pdf"
+
+      inserted = []
+      Prawn::Document.generate(stamp_path, margin: [0, 0]) do |pdf|
+        y_representative_coords = page == 1 ? 118 : 216
+        y_veteran_coords = page == 1 ? 90 : 322
+        pdf.image representative_signature, at: [35, y_representative_coords], height: 20
+        inserted << 'representative'
+        pdf.image veteran_signature, at: [35, y_veteran_coords], height: 20
+      end
+      stamp(pdf_path, stamp_path, delete_source: false)
+    rescue Prawn::Errors::UnsupportedImageType
+      signature = inserted.empty? ? 'representative' : 'veteran'
+      update signature_errors: ["#{signature} was not a recognized image format"]
```

Comment:
The errors stored here should probably use our error model for claims 

```
{
  "errors": [
    {
      "status": "422",
      "source": "#/serviceInformation/servicePeriods/1/serviceBranch",
      "details": "burial is not currently supported, but will be in a future version"
    }
  ]
}
```
Or at the very least should include the source as the signature block and the description so it follows convention. I do like the differentiation between veteran and representative signatures though, nice touch.

Reply:
We are also not exposing the errors back to the user in the status check somewhere when they look up this GUID but that might merit a different Ticket in this case.

Reply:
I think its a different ticket because we also arent even exposing the generated pdf either.

Reply:
Also, these errors are coming from json schema, but we dont actually know if we have an error until the job process the base64 into a png